src/Mod/Part/Gui/SectionCutting.cpp

```diff
@@ -0,0 +1,1479 @@
+        return;
+    }
+    // empty the ObjectsListVisible
+    ObjectsListVisible.clear();
+    // now store those that are currently visible
+    for (auto it = ObjectsList.begin(); it != ObjectsList.end(); ++it) {
+        if ((*it)->Visibility.getValue())
+            ObjectsListVisible.push_back((*it));
+    }
+
+    // we can have existing cut boxes take their values
+    // the flip state cannot be readout of the box position, therefore readout the position
+    // is if it was unflipped
+    if (doc->getObject(BoxZName)) {
+        hasBoxZ = true;
+        ui->groupBoxZ->setChecked(true);
+        Part::Box* pcBox = static_cast<Part::Box*>(doc->getObject(BoxZName));
+        if (!pcBox) {
+            Base::Console().Error("SectionCut error: cut box is incorrectly named, cannot proceed\n");
+            return;
+        }
+        ui->cutZ->setValue(pcBox->Height.getValue() - fabs(pcBox->Placement.getValue().getPosition().z));
+    }
+    if (doc->getObject(BoxYName)) {
+        hasBoxY = true;
+        ui->groupBoxY->setChecked(true);
+        Part::Box* pcBox = static_cast<Part::Box*>(doc->getObject(BoxYName));
+        if (!pcBox) {
+            Base::Console().Error("SectionCut error: cut box is incorrectly named, cannot proceed\n");
+            return;
+        }
+        ui->cutY->setValue(pcBox->Width.getValue() - fabs(pcBox->Placement.getValue().getPosition().y));
+    }
+    if (doc->getObject(BoxXName)) {
+        hasBoxX = true;
+        ui->groupBoxX->setChecked(true);
+        Part::Box* pcBox = static_cast<Part::Box*>(doc->getObject(BoxXName));
+        if (!pcBox) {
+            Base::Console().Error("SectionCut error: cut box is incorrectly named, cannot proceed\n");
+            return;
+        }
+        ui->cutX->setValue(pcBox->Length.getValue() - fabs(pcBox->Placement.getValue().getPosition().x));
+    }
+
+    // hide existing cuts to check if there are objects to be cut visible
+    if (doc->getObject(CutXName))
+        doc->getObject(CutXName)->Visibility.setValue(false);
+    if (doc->getObject(CutYName))
+        doc->getObject(CutYName)->Visibility.setValue(false);
+    if (doc->getObject(CutZName))
+        doc->getObject(CutZName)->Visibility.setValue(false);
+
+    // get bounding box
+    SbBox3f box = getViewBoundingBox();
+    if (!box.isEmpty()) {
+        // if there is a cut box, perform the cut
+        if (hasBoxX || hasBoxY || hasBoxZ) {
+            // refresh only the range since we set the values above already
+            refreshCutRanges(box, Refresh::notXValue, Refresh::notYValue, Refresh::notZValue,
+                Refresh::XRange, Refresh::YRange, Refresh::ZRange);
+        }
+        else
+            refreshCutRanges(box);
+    }
+    // the case of an empty box and having cuts will be handles later by startCutting(true)
+
+    connect(ui->groupBoxX, SIGNAL(toggled(bool)), this, SLOT(onGroupBoxXtoggled(bool)));

```

Comment:
I suggest migrating this to the non-macro versions so that you get type-checking. So, e.g. something like
```suggestion
    connect(ui->groupBoxX, &QGroupBox::toggled, this, &SectionCut::onGroupBoxXtoggled);
```

Reply:
Thanks.

Done now. I see that I am the first one in the whole code using 
`QGroupBox::toggled`

I did what you suggested but I am not sure I did it right for QDoubleSpinBox. I found this:
https://doc.qt.io/qt-5/qdoublespinbox.html#valueChanged
Can you please have a Look if I implemented this correctly?

Reply:
It looks right to me: the nice thing about the new style is that if you are doing it wrong you get a compiler error instead of having to wait for it to fail at runtime. So mostly if it compiles it's correct. I have not actually tested the interface yet, but I think you've got the code alright now.

Reply:
Thanks. Yes, it compiles without warnings, the CI run fine and I checked that all intended functionality is there.

Since the UI was in the making for more than a year now, it is now time to go live. -> merging